manifests/init.pp

```diff
@@ -101,4 +101,13 @@
     environment => concat([ "VENV_PATH=${venv_path}" ], $environment),
     refreshonly => true,
   }
+
+  # Used in letsencrypt::certonly Exec["letsencrypt certonly ${title}"]
+  file { '/usr/local/sbin/letsencrypt-domain-validation':
+    ensure => file,
+    owner  => 'root',
+    group  => 'root',
+    mode   => '0500',
```

Comment:
Any reason not to use 0755 or even 0777?

Reply:
May I answer this with a counter question? :) Any reason not to use '0500'?

We usually do this for scripts which are only used in Puppet which I find a good practice given that no other user should use that script. Its basically a simple `assert_private()` for scripts :D